### DISCUSSION TITLE
How to handle gzip compressed results?

### DISCUSSION THREAD

**Discussion by BuzzDee3000** [canton7/RestEase#195] [Q&A] [answered]
Hi, 
we are using RestEase and are very happy with it. 
Now we implement a new external API and see that RestEase is not handling the compressed gzip response within default behavior. 

Is there a way to let RestEase handle the decompression?

**Comment by canton7** [comment, marked as answer]
This is an HttpClient-level concern. Set `HttpClientHandler.AutomaticDecompression Property = DecompressionMethods.All`.

(This was the default for a while, but annoyingly was reverted back to `None` for backwards compat concerns).

```cs
var httpClientHandler = new HttpClientHandler()
{
    AutomaticDecompression = DecompressionMethods.All,
};
var httpClient = new HttpClient(httpClientHandler)
{
    BaseAddress = new Uri(....)
};
var client = RestClient.For<ISomeApi>(httpClient);
```

**Reply by BuzzDee3000** [reply]
Thank you for this quick response. 
What about something inside RestEase to configure this?

**Reply by canton7** [reply]
Yeah, it wouldn't be the worst thing in the world to just set it as a courtesy to the user. I wouldn't want to change it if the user passes in their own `HttpClient` though, and that means there's a potential gotcha hiding there.
